src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -49,14 +49,14 @@ private void CheckForInvalidStack (Stack<StackSlot> stack, int depthRequired, Me
 			if (stack.Count < depthRequired) {
 				WarnAboutInvalidILInMethod (method, ilOffset);
 				while (stack.Count < depthRequired)
-					stack.Push (new StackSlot ()); // Push dummy values to avoid crashes.
-												   // Analysis of this method will be incorrect.
+					stack.Push (new StackSlot (UnknownValue.Instance)); // Push dummy values to avoid crashes.
+												                        // Analysis of this method will be incorrect.
 			}
 		}
 
 		private static void PushUnknown (Stack<StackSlot> stack)
 		{
-			stack.Push (new StackSlot ());
+			stack.Push (new StackSlot (UnknownValue.Instance));
```

Comment:
One would think that the nullable analysis helped me to find these spots, but today I learned that:

```csharp
Foo f = new();
f.Obj.ToString();

struct Foo
{
    public object Obj { get; }
    public Foo(object obj) => Obj = obj;
}
```

will crash with a NullReferenceException, `#nullable enabled` be damned.

Reply:
Yeah - nullable on structs is... a lie :wink:. The default .ctor on struct is just evil - in many ways.

Reply:
We could declare a parameterless .ctor on the struct and make it throw. It doesn't catch all cases (the default initialization will not invoke it), but it would catch all the `new StackSlot()` calls.